### DISCUSSION TITLE
[Request] Non-censored secrets/dynamic variables

### DISCUSSION THREAD

**Discussion by nihaals** [community/community#26055] [Actions]
<p>This is something I’ve wanted for a while but seems even more useful now due to the Docker Hub pull limit changes (see <a href="https://github.community/t/did-dockerhub-rate-limit-affect-github-action/128158">this post</a>, I’m mainly referring to the linked <a href="https://github.com/actions/virtual-environments/issues/1445" rel="nofollow noopener">GitHub issue</a>).</p>
<p>Many people input their Docker repository username as a secret. This can be useful as you can:</p>
<ol>
<li>Change them without committing a change (especially useful if it’s more urgent and you have a PR policy)</li>
<li>Censor it if it’s an email or another random account (e.g. a designated CI account/token)</li>
</ol>
<p>But, when you’re doing something like maintaining a library, you are often using Docker Hub and the same username to push (and even pull in the future) as the Docker Hub repository owner. Using a secret results in unnecessary censoring which could make logs quite difficult to read. If you end up switching to a designated account or use something like Google Container Registry, you would want to switch to a secret as the username may be sensitive, but the only way to do both of these things is to initially have it hard coded into the workflow (e.g. an env var) then commit a change switching it to a secret. Initially using a hard coded value loses the first advantage mentioned above.</p>
<p>I was initially thinking of a separate settings-like tab where you could enter variables (which wouldn’t simply be passed into every workflow and step and instead behave similarly to secrets), but having a toggle in the add secret input would make more sense, be less confusing, make it more visible and have it be very similar to an existing setup (no referencing non-censored variables as a secret in the workflow).</p>
<p>Having a tick box when adding/editing a secret which marks it as non-sensitive (meaning it is not censored in logs) along with some indication in the secrets list that it is not censored (and maybe even showing the existing value when editing it/allowing users to see the value of it without resetting it) would help make logs more readable in some cases and allow for the benefits of secrets to be used for more things. There are probably more examples than Docker usernames which would also benefit from this feature.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/nihaals">@nihaals</a>,</p>
<p>Thanks for your feedback.</p>
<p>GitHub takes your suggestions very seriously, and the suggestions are very helpful for improving GitHub.</p>
<p>I recommend that you can directly share your suggestion <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">here</a>. That will allow you to directly interact with the appropriate engineering team, and make it more convenient for the engineering team to collect and categorize your suggestions.</p>

**Comment by nihaals** [comment]
<p>Thanks, I’ve submitted it there. Out of interest, was this policy changed recently? I wanted to avoid posting this in the wrong place but <a href="https://github.community/t/feature-request-include-pull-request-reviews-information-with-pull-request-data-on-the-api/13983/2">this post</a> suggested posting on the forums is fine too</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/nihaals">@nihaals</a>,</p>
<p>About where you need to post your topics, you can reference “<a href="https://docs.github.com/en/actions/getting-started-with-github-actions/about-github-actions#contacting-support" rel="nofollow noopener">Contacting support</a>”.</p>
